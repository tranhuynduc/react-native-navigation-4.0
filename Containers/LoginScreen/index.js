import React, { useState } from 'react';
import { View, Text, Button, TextInput } from 'react-native';
import {
  Container,
  StyledLayout,
  FormControl,
  FormGroup,
  FormLabel,
  StyledButton,
  Paragraph,
} from '../../styled/Layout';
import { TouchableOpacity } from 'react-native-gesture-handler';

const LoginScreen = ({ navigation }) => {
  const [username, setUsername] = useState('Admin');
  const [password, setPassword] = useState('123');
  const [error, setError] = useState('');

  const login = () => {
    if (!username) {
      return setError('Please input your username');
    }

    if (!password) {
      return setError('Please input your password');
    }

    navigation.navigate('Statistic');
  };
  return (
    <Container>
      <FormGroup>
        <FormLabel>Username</FormLabel>
        <FormControl
          placeholder="Input your username"
          onChangeText={text => setUsername(text)}
          value={username}
        />
      </FormGroup>
      <FormGroup>
        <FormLabel>Password</FormLabel>
        <FormControl
          placeholder="Input your password"
          onChangeText={text => setPassword(text)}
          value={password}
        />
      </FormGroup>
      <StyledButton color="primary" onPress={() => login()}>
        <Paragraph color="white">Login</Paragraph>
      </StyledButton>
      {error !== '' && (
        <Paragraph color="danger" align="left">
          {error}
        </Paragraph>
      )}
    </Container>
  );
};

export default LoginScreen;
