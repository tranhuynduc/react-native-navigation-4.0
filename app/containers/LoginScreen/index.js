import React, { useState } from 'react';
import { Button, Card, Paragraph, TextInput } from 'react-native-paper';
import { Spacing } from '../../styled/Layout';
import DismissKeyboard from '../../components/DismissKeyboard';
class LoginScreen extends React.Component {
  static navigationOptions = {
    noMenu: true,
  };

  constructor() {
    super();
    this.state = {
      username: 'Admin',
      password: '123',
      error: '',
    };
  }

  handleLogin = () => {
    const { username, password } = this.state;
    const { navigation } = this.props;
    if (!username) {
      return this.setState({ error: 'Please input your username' });
    }

    if (!password) {
      return this.setState({ error: 'Please input your password' });
    }

    navigation.navigate('Statistic');
  };

  handleChangeUsername = text => this.setState({ username: text });
  handleChangePassword = text => this.setState({ password: text });

  render() {
    const { username, password, error } = this.state;
    return (
      <DismissKeyboard>
        <Card style={{ flex: 1 }}>
          <Card.Content
            style={{
              flex: 1,
              alignContent: 'center',
              justifyContent: 'center',
            }}>
            <TextInput
              label="Username"
              mode="outlined"
              placeholder="Input your username"
              value={username}
              onChangeText={this.handleChangeUserName}
              data-test="input-username"
            />
            <TextInput
              label="Password"
              mode="outlined"
              value={password}
              onChangeText={this.handleChangePassword}
              data-test="input-password"
            />
            <Spacing />
            <Button icon="account" mode="contained" onPress={this.handleLogin}>
              Login
            </Button>
            <Paragraph>{error}</Paragraph>
          </Card.Content>
        </Card>
      </DismissKeyboard>
    );
  }
}

export default LoginScreen;
