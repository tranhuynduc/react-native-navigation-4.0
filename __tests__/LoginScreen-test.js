/**
 * @format
 */

import 'react-native';
import React from 'react';
import { shallow, mount } from 'enzyme';
import LoginScreen from '../app/containers/LoginScreen';
import renderer from 'react-test-renderer';
export const findByTestAttr = (wrapper, val) => {
  return wrapper.find(`[data-test="${val}"]`);
};

describe('Login Screen', () => {
  const props = {
    navigation: { navigate: jest.fn() },
  };
  let wrapper = null;
  let instaneOf;
  beforeEach(() => {
    wrapper = shallow(<LoginScreen {...props} />);
    instaneOf = renderer.create(<LoginScreen {...props} />).getInstance();
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state username/password when input change', () => {
    const usernameValue = 'username';
    const passwordValue = 'password';
    instaneOf.handleChangeUsername(usernameValue);
    instaneOf.handleChangePassword(passwordValue);
    expect(instaneOf.state.username).toEqual(usernameValue);
    expect(instaneOf.state.password).toEqual(passwordValue);
  });

  it('should show error when user does not input username', () => {
    instaneOf.setState({ username: '' });
    instaneOf.handleLogin();
    expect(instaneOf.state.error).not.toEqual('');
  });

  it('should show error when user inputs username but does not input password', () => {
    instaneOf.setState({ username: 'admin', password: '' });
    instaneOf.handleLogin();
    expect(instaneOf.state.error).not.toEqual('');
  });

  it('should call navigate if user input username and password', () => {
    instaneOf.setState({ username: 'admin', password: 'password' });
    instaneOf.handleLogin();
    expect(props.navigation.navigate).toHaveBeenCalled();
  });
});
