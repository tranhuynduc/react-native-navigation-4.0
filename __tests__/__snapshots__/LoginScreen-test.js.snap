// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Screen renders correctly 1`] = `ShallowWrapper {}`;
